src/recordings.js

```diff
@@ -11,3 +13,84 @@ export async function fetchFirstRecordingURL(wordform) {
+  let xhttp = new XMLHttpRequest();
+  xhttp.open('GET', derivedURL, true);
+
+  // receiving request information from SAPIR
+  xhttp.onload = function() {
+    let returnedData = JSON.parse(this.response); // response from the server
+    let numberOfRecordings = returnedData.length; // number of records on the server
+    let recordingsList = document.querySelector('.recordings-list');
+    let recordingsHeading = document.querySelector('.definition__recordings');
+    
+    // we only want to display our list of speakers once!
+    if (recordingsList.childElementCount < numberOfRecordings) {
+      displaySpeakerList(returnedData);
+    }
+
+    // the function that displays an individual speaker's name
+    function displaySpeakerList(firstJSONData) {
+      recordingsHeading.insertAdjacentHTML('afterbegin', '<h3 class="explainer">Select a name below to hear the word above said by different speakers.</h3>');
+      
+      let speakerURLIndexCount = 0;
+
+      while (speakerURLIndexCount < numberOfRecordings) {
+        // create a list element and set an attribute on it for testing
+        let individualSpeaker = document.createElement('li');
+        individualSpeaker.classList.add('recordings-list__item');
+        individualSpeaker.setAttribute('data-cy', 'recordings-list__item');
+
+        // create a button element: add a class to it for future styling needs
+        let speakerButton = document.createElement('button');
+        speakerButton.classList.add('audio-snippet');
+
+        // put the button into the list
+        individualSpeaker.appendChild(speakerButton);
+
+        // put the list into the DOM
+        recordingsList.appendChild(individualSpeaker);
+
+        speakerURLIndexCount++;
+      }
+
+      // TODOkobe: hey future Eddie (+ Kobe), should the for-loop be within the while loop above?
+
+      /**
+      * Add text to the newly created buttons with a for-loop and get audio playback for each button
+      */
+      for (let speakerURLIndexCount = 0; speakerURLIndexCount < firstJSONData.length; speakerURLIndexCount++) {
+
+        // select for the buttons...
+        let createdSpeakerButton = document.querySelectorAll('button.audio-snippet');
+
+        // ...and then iterate through them to add text
+        createdSpeakerButton[speakerURLIndexCount].innerText = firstJSONData[speakerURLIndexCount].speaker_name + ', Maskwacîs';
+
+        // put an event listener on the button: the event is the URL playback
+        createdSpeakerButton[speakerURLIndexCount].addEventListener('click', function() {
+          var audio = new Audio(firstJSONData[speakerURLIndexCount].recording_url);
+          audio.type = 'audio/m4a';
+          audio.play();
+
+          // what i want to do is grab the JSON and make an array based on the speaker_url property. with that, I want to grab the things associated with it – the speaker name, namely. i also want to keep the position of the speaker_urls as they are.
```

Comment:
Is this comment still relevant? If not, ✂️ 
```suggestion
```

Reply:
Oop. Yes, we can nix this!